crates/astria-composer/src/searcher/api.rs

```diff
@@ -0,0 +1,58 @@
+use std::net::SocketAddr;
+
+use axum::{
+    extract::State,
+    response::IntoResponse,
+    routing::get,
+    Router,
+};
+use serde::Serialize;
+use thiserror::Error;
+
+use super::State as SearcherState;
+
+#[derive(Debug, Error)]
+pub enum ApiError {
+    #[error("hyper error")]
+    ServerError(#[from] hyper::Error),
+}
+
+pub(super) async fn run(api_url: &SocketAddr, state: &SearcherState) -> Result<(), ApiError> {
+    let api_router = Router::new()
+        .route("/healthz", get(healthz))
+        .with_state(state.clone()); // TODO: should use an Arc here instead of cloning, otherwise state is stale
+
+    axum::Server::bind(api_url)
+        .serve(api_router.into_make_service())
+        .await
+        .map_err(ApiError::ServerError)
```

Comment:
You don't need to wrap that error here. If you have several long-running tasks that are at the same "level", you can just spawn all of them into their respective tasks and `tokio::select!` or `tokio::join!` on them.

In fact, I did something like that here: https://github.com/astriaorg/astria/pull/89/files#diff-79f07f63704e1b426ed89a94ebf5327ed84d4f64dd76efb420b8f84eaeabe0bc

(still need to merge it)